lms/static/scripts/frontend_apps/components/Dialog.js

```diff
@@ -42,48 +43,37 @@ export default function Dialog({
   title,
   buttons,
 }) {
-  const handleKey = event => {
-    event.stopPropagation();
-
-    if (event.key === 'Escape' && typeof onCancel === 'function') {
+  const contentEl = useRef(null);
+  useElementShouldClose(contentEl, true, () => {
+    // onCancel callback prop is optional
+    if (typeof onCancel === 'function') {
       onCancel();
     }
-  };
+  });
 
-  const rootEl = useRef(null);
   useEffect(() => {
     if (initialFocus) {
       initialFocus.current.focus();
-    } else {
-      // Modern accessibility guidance is to focus the dialog itself rather than
-      // trying to be smart about focusing a particular control within the
-      // dialog. See resources above.
-      rootEl.current.focus();
     }
-
     // We only want to run this effect once when the dialog is mounted.
     //
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   return (
-    // FIXME-A11Y
-    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
     <div
-      role="dialog"
+      role="alertdialog"
       aria-labelledby="Dialog__title"
-      onKeyDown={handleKey}
-      // FIXME-A11Y
-      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-      tabIndex="0"
-      ref={rootEl}
+      aria-modal="true"
+      aria-describedby="Dialog__content"
```

Comment:
Hmm, is it common to use `aria-describedby` for the entire content of the dialog? Also, I'd suggest that it probably would be a good idea to adopt a different syntax for IDs as opposed to class names to avoid mixing them up (eg. where someone adds an `aria-` attribute but then sets a class rather than an ID on the target element).

Reply:
I think since the content is fully customized, it may not be valuable to even add aria-describedby, and perhaps the aria-labelledby is enough. Thoughts?

I'll wait on the other PR to land which deals with the ids, then I'll update this one again.